cterasdk/edge/ssl.py

```diff
@@ -37,31 +40,57 @@ def _get_force_https(self):
     def _set_force_https(self, force):
         self._gateway.put('/config/fileservices/webdav/forceHttps', force)
 
-    def _load_cert(self, certificate, private_key, reboot=True, wait_for_reboot=False):
-        logging.getLogger().info("Uploading server certificate and private key")
-        server_certificate = '\n' + private_key + certificate
-        response = self._gateway.put('/config/certificate', server_certificate)
-        logging.getLogger().info("Uploaded server certificate and private key.")
-        if reboot:
-            self._gateway.power.reboot(wait_for_reboot)
-        return response
+    def remove_storage_ca(self):
+        """
+        Remove object storage trusted CA certificate
+        """
+        self._gateway.put('/config/extStorageTrustedCA', None)
 
-    def upload_cert(self, certificate, private_key, reboot=True, wait_for_reboot=False):
+    def get_storage_ca(self):
         """
-        Upload a server certificate
+        Get object storage trusted CA certificate
+        """
+        return self._gateway.get('/status/extStorageTrustedCA')
+
+    def set_storage_ca(self, certificate):
+        """
+        Set object storage trusted CA certificate
 
-        :param str certificate: A path to the PEM-encoded server certificate file
-        :param str private_key: A path to the PEM-encoded private key
+        :param str certificate: The PEM-encoded certificate or a path to the PEM-encoded server certificate file
         """
-        file_info, cert = SSL._file_contents(certificate)
-        logging.getLogger().debug(
-            {'name': file_info['name'], 'size': file_info['size'], 'type': file_info['mimetype']}
-        )
-        file_info, pk = SSL._file_contents(private_key)
-        logging.getLogger().debug(
-            "Read private key file. %s", {'name': file_info['name'], 'size': file_info['size'], 'type': file_info['mimetype']}
-        )
-        return self._load_cert(cert, pk, reboot, wait_for_reboot)
+        logging.getLogger().info('Setting trusted object storage CA certificate')
+        param = Object()
+        param._classname = 'ExtTrustedCA'  # pylint: disable=protected-access
+        param.certificate = SSL._obtain_secret(certificate)
+        return self._gateway.put('/config/extStorageTrustedCA', param)
+
+    def set_certificate(self, private_key, *certificates):
+        """
+        Set the Edge Filer's web server's certificate.
+
+        :param str private_key: The PEM-encoded private key or a path to the PEM-encoded private key file
+        :param list[str] certificates: The PEM-encoded certificates or a path to the PEM-encoded certificates
+        """
+        certificate_chain = []
+        logging.getLogger().debug('Loading private key')
+        certificate_chain.append(SSL._obtain_secret(private_key))
```

Comment:
Just initialize the array with the private key. No need to initialize an empty array and then append

Reply:
Done